app/Http/Controllers/PostsController.php

```diff
@@ -175,12 +175,14 @@ public function update(PostRequest $request, Post $post)
         // Only allow an admin/staff or the user who owns the post to update.
         $this->authorize('update', $post);
 
+        $validatedRequest = $request->validated();
+
         // But don't allow user's to review their own posts.
         if (! Gate::allows('review', $post)) {
-            unset($request['status']);
+            unset($validatedRequest['status']);
         }
 
-        $this->posts->update($post, $request->validated());
+        $this->posts->update($post, $validatedRequest);
```

Comment:
This was a funky bug. Some of the logic and return values for validation in 5.7 have changed a bit and `$request->validated()` was causing the previously unset `status` array item in the `Gate` conditional above to be popped back into the array, causing the test to fail for post owner's not being able to review their own post.

So I revised a logic order a bit and call `$request->validated()` before the `Gate` check, so the final `$validatedRequest` array passed to the `update()` method only contains the correct, allowed items.

Although I think this could be simplified to just pass through the request array as:

```php
$this->posts->update($post, $request->toArray());
```

I didn't change it to the simpler approach above in case I was missing something with _why_ we need to use the `validated()` method, but given that this is data passed through the `PostRequest` it should already be valid, and not sure what `validated()` is adding (especially with the [changes in 5.7](https://github.com/laravel/framework/blob/2555bf6ef6e6739e5f49f4a5d40f6264c57abd56/src/Illuminate/Validation/Validator.php#L312-L319) where when `validate()` is called, the data is sent through the `validated()` method at the end of the process as well. 🤷‍♂️

Reply:
Oh wow, glad we had a test to cover this! :relieved: 

The reason we use the returned `$validatedRequest` is because it trims out any additional properties not covered by the [validator](https://github.com/DoSomething/rogue/blob/8bcfe954aa20233afe0377cedd45d5a03bb218de/app/Http/Requests/PostRequest.php) (sort-of like how Mass Assignment protection works, but we actually use it)! So for example, if someone included `diego => 'hacker'` in their post payload, it'd be excluded from `$validatedRequest`.